internal/provider/search_attribute_resource.go

```diff
@@ -0,0 +1,292 @@
+		)
+		return
+	}
+
+	r.client = client
+	tflog.Info(ctx, "Configured Temporal Search Attribute client", map[string]any{"success": true})
+}
+
+// Create is responsible for creating a new search attribute in Temporal.
+func (r *SearchAttributeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data SearchAttributeResourceModel
+
+	client := operatorservice.NewOperatorServiceClient(r.client)
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create attribute
+	indexedValueType, _ := enums.IndexedValueTypeFromString(data.Type.ValueString())
+
+	request := &operatorservice.AddSearchAttributesRequest{
+		Namespace:        data.Namespace.ValueString(),
+		SearchAttributes: map[string]enums.IndexedValueType{data.Name.ValueString(): indexedValueType},
+	}
+	_, err := client.AddSearchAttributes(ctx, request)
+
+	if err != nil {
+		if _, ok := err.(*serviceerror.AlreadyExists); ok {
+			resp.Diagnostics.AddError("Request Error", "Search attribute with that name is already registered: "+err.Error())
+			return
+		}
+		resp.Diagnostics.AddError("Request error", "Search attribute creation failed: "+err.Error())
+		return
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("The search attribute: %s of type %s is successfully created", data.Name, data.Type.String()))
+	tflog.Trace(ctx, "Created a resource")
+
+	// Verify attribute creation
+	attributes, err := client.ListSearchAttributes(ctx, &operatorservice.ListSearchAttributesRequest{
+		Namespace: data.Namespace.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read search attribute info, got error: %s", err))
+		return
+	}
+
+	_, ok := attributes.CustomAttributes[data.Name.ValueString()]
+	if !ok {
+		resp.Diagnostics.AddError("Verification Error", fmt.Sprintf("%s not found, creation could not be verified", data.Name.ValueString()))
+		return
+	}
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read is responsible for reading the current state of a Temporal search attribute.
+// It fetches the current configuration of the search attribute and updates the Terraform state.
+func (r *SearchAttributeResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state SearchAttributeResourceModel
+
+	client := operatorservice.NewOperatorServiceClient(r.client)
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	attributes, err := client.ListSearchAttributes(ctx, &operatorservice.ListSearchAttributesRequest{
+		Namespace: state.Namespace.ValueString(),
```

Comment:
What if this value is empty?

Reply:
Search attribute resource already has a namespace default `default`, as indicated in the schema so this won't be empty